File: college/lista-aula-3/questao-4.js
const prompt = require('prompt-sync')({ sigint: true });

let totalCobaias = 0;
// cod 1
let totalCoelhos = 0;
// cod 2
let totalRatos = 0;

console.log('Bem vindo à calculadora de cobaias!');

for (let i = 1; i <= 15; i++) {
  let tipo = 0;
  while (true) {
    console.log('');
    tipo = Number(
      prompt(
        `Digite o tipo de cobaia utilizada na experiência ${i} (1 para coelhos e 2 para ratos): `
      )
    );

    if (tipo === 1 || tipo === 2) break;

    console.log('Entrada inválida.');
  }

  const qtdCobaias = Number(prompt('Digite a quantidade de cobaias: '));

  totalCobaias += qtdCobaias;

  if (tipo === 1) totalCoelhos += qtdCobaias;
  if (tipo === 2) totalRatos += qtdCobaias;
}

console.log(`Na experiência foram utilizadas ${totalCobaias} cobaias.`);
console.log(`Na experiência foram utilizadas ${totalCoelhos} coelhos.`);
console.log(`Na experiência foram utilizadas ${totalRatos} ratos.`);

console.log(`Foram ${(totalCoelhos/totalCobaias * 100).toFixed(2)}% de coelhos e ${(totalRatos/totalCobaias * 100).toFixed(2)}% de ratos.`)

File: python/lista-3-facul/questao-4.py
totalCobaias = 0;
totalCoelhos = 0;
totalRatos = 0;

print('Bem vindo à calculadora de cobaias!');

for i in range(1, 4):
  while True:
    tipo = input(f'Digite o tipo de cobaia utilizada na experiência {i} (1 para coelhos e 2 para ratos): ');

    if tipo.isnumeric():
      if int(tipo) == 1: 
          qtdCobaias = int(input('Digite a quantidade de cobaias: '));
          totalCoelhos += qtdCobaias;
          totalCobaias += qtdCobaias;
          break;
      elif int(tipo) == 2: 
        qtdCobaias = int(input('Digite a quantidade de cobaias: '));
        totalRatos += qtdCobaias;
        totalCobaias += qtdCobaias;
        break;
      else:
        print('Entrada inválida.');
    else:
      print('Entrada inválida.');

else:
  print(f'Na experiência foram utilizadas {totalCobaias} cobaias.')
  print(f'Na experiência foram utilizadas {totalCoelhos} coelhos.')
  print(f'Na experiência foram utilizadas {totalRatos} ratos.')

  print(f'Foram {(totalCoelhos/totalCobaias * 100):.2f}% de coelhos e {(totalRatos/totalCobaias * 100):.2f}% de ratos.')
